app/src/ui/blank-slate/create-tutorial-repository-dialog.tsx

```diff
@@ -148,6 +151,8 @@ export class CreateTutorialRepositoryDialog extends React.Component<
   }
 
   public onSubmit = async () => {
+    this.props.dispatcher.recordTutorialStarted()
```

Comment:
Is the intention behind having both `tutorialStarted` and `tutorialRepositoryCreated` to be able to differentiate users where the tutorial repository creation fails? If it is I feel like a dedicated metric would be more understandable and if it isn't I wonder if we could make do with just one of them given that they both get set during creation.

Furthermore, if we could make do with just one of them I think we should be able to call that method from the `onNewTutorialRepository` method on the tutorial assessor and thereby avoid passing the dispatcher to the `CreateTutorialRepositoryDialog` component entirely (and also get rid of the two dispatcher methods).

Reply:
@niik yup, you've got the intention. My logic was if we have attrition between the user clicking the button and completing the first step, we'd want to know how many we're losing due to repo creation versus editor installation. I was imagining a sort of waterfall chart that shows the fall-off from clicking the button (tutorialStarted) to getting to the welcome page/install editor step (tutorialRepoCreated) to getting to the second step (tutorialEditorInstalled), etc. 

All that said, what your describing as an alternative also sounds reasonable -- a dedicated metric for repo creation failure plus just one of either tutorialStarted or tutorialRepoCreated, and dropping the dispatcher methods. From an implementation standpoint that might be cleaner. From a metric-interpreting standpoint, for some reason my mind finds the current implementation more straight-forward/easier to reason about. 

That said, I don't have a strong preference either way. Nor does @outofambit. We discussed things for a bit, and ultimately decided to leave the call up to you. Please feel free to make changes as you see fit. We thought about cutting a branch and guessing at the changes you suggested to give you options to choose from, but ran out of time to implement them, and figured it (hopefully) wouldn't take you too long to make the changes you want (and they'd be exactly what you want rather than our interpretation). Whatever you decide on, we support you :)